core/kazoo_token_buckets-1.0.0/src/kz_token_bucket_pqc.erl

```diff
@@ -0,0 +1,289 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz INC
+%%% @doc
+%%%
+%%% The goal is to model how a token bucket works and, using PropEr,
+%%% generate test scenarios that prove the properties.
+%%%
+%%% Our model is a bucket of tokens, starting (and capped) at MAX,
+%%% with a refill rate of FR per unit time FRT. We will represent this
+%%% as a tuple {TOKENS, MAX, FR, FRT}.
+%%%
+%%% We want to test that consuming tokens works as long as the requested
+%%% amount is < TOKENS, that the bucket refills at FR/FRT, and that the
+%%% bucket cannot exceed MAX tokens.
+%%%
+%%% @end
+%%% @contributors
+%%%   James Aimonetti
+%%%-------------------------------------------------------------------
+-module(kz_token_bucket_pqc).
+
+-ifdef(TEST).
+
+-include("kz_buckets.hrl").
+
+-include_lib("proper/include/proper.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-behaviour(proper_statem).
```

Comment:
We used to include proper, but since it was only for testing and is licensed as GPL, there were concerns with including it. So we stopped including it. This will only run if you exec 'make test' so I would assume you'd have proper in your Erlang load path. I have it in among the other core Erlang/OTP libs myself.